### PR TITLE
upgrading: initiating upgrades with the UI

### PR DESCRIPTION
**Pull request by ghost**
Part 3 of the upgrrading series: Initiating upgrades from the UI.

You can test it here: https://www.katacoda.com/drsk/courses/upgrading/upgrade-ui

### CODE REVIEW

File: upgrading/upgrade-ui/init_background.sh [hunk 1]
```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+
+echo initializing...
+
+# workaround for assets in the `assets` directory, because google-chrome complains when the file
+# size is bigger than 50kb
+wget https://katacoda.com/drsk/courses/upgrading/upgrade-ui/assets/create-daml-app.tar.gz
```
Comment (nemanja-da) [inline upgrading/upgrade-ui/init_background.sh]:
```suggestion
wget https://katacoda.com/daml/courses/upgrading/upgrade-ui/assets/create-daml-app.tar.gz
```

Let's use the prod link (I know drsk one was needed for testing)

File: upgrading/upgrade-ui/1_codegen.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
+We start with the following setup: Your social network of `create-daml-app-0.1.0` and `forum-0.1.0`
+are running and your users have been happily posting messages in the forum. You decide to upgrade to
```
Comment (nemanja-da) [inline upgrading/upgrade-ui/1_codegen.md]:
```suggestion
We start with the following setup (currently being set up in terminal 1): Your social network of `create-daml-app-0.1.0` and `forum-0.1.0`
are running and your users have been happily posting messages in the forum. You decide to upgrade to
```

**Review by nemanja-da — APPROVED** [review]
Thnx @robin-da Will add some minor comments later

File: upgrading/upgrade-ui/1_codegen.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+`create-daml-app-0.1.1` and hence you write a migration package `migration-0.1.0` and deploy it
+together with the new packages `create-daml-app-0.1.1`, `forum-0.1.1`.
+
+Now you want to drive the upgrade process of the `create-daml-app-0.1.0` DAML application from the
+UI.  If a user wants to login, whose `User` contract is still of the old `create-daml-app-0.1.0`
+package, you ask her kindly to upgrade to the newer package. 
+
+In order to do so, you need the generated JavaScript bindings for the newly available packages
+available. **Wait for the sandbox to boot in the background**. Then create them with the DAML
```
Comment (nemanja-da) [inline upgrading/upgrade-ui/1_codegen.md]:
```suggestion
In order to do so, you need the generated JavaScript bindings for the newly available packages. **Wait for the sandbox to boot in the background**. Then create them with the DAML
```

File: upgrading/upgrade-ui/1_codegen.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+```
+daml codegen js create-daml-app-0.1.0.dar -o daml.js
+daml codegen js create-daml-app-0.1.1.dar -o daml.js
+daml codegen js forum-0.1.0.dar -o daml.js
+daml codegen js migration-0.1.0.dar -o daml.js
+```{{execute T1}}
+
+Once finished, open `ui/package.json`{{open}} in the IDE and add the new dependencies of
+`create-daml-app-0.1.1`, `forum-0.1.0` and `migration-0.1.0`
```
Comment (nemanja-da) [inline upgrading/upgrade-ui/1_codegen.md]:
```suggestion
Once finished, open `ui/package.json`{{open}} in the IDE and add the new dependencies for the
`create-daml-app-0.1.1`, `forum-0.1.0` and `migration-0.1.0` javascript packages
```

File: upgrading/upgrade-ui/1_codegen.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+and install them with
+
+```
+cd ui
+yarn install
+```{{execute T1}}
+
+Notice that we renamed the `create-daml-app-0.1.0` import from `create-daml-app` to
+`create-daml-app-0.1.0`, while we import `create-daml-app-0.1.1` now as `create-daml-app`. This
```
Comment (nemanja-da) [inline upgrading/upgrade-ui/1_codegen.md]:
```suggestion
Notice that `create-daml-app` import is now pointing to `create-daml-app-0.1.1`. We point to the old old contracts and its bindings explicitly via `create-daml-app-0.1.0` import.
```
Had to read the text couple of times, the above suggestion made sense to me. Is it import or dependency? What wording is more appropriate here?

File: upgrading/upgrade-ui/2_ui.md [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+`create-daml-app-0.1.0`, `forum-0.1.0` and `migrate-0.1.0` JavaScript bindings libraries:
+
+<pre class="file" data-target="clipboard">
+import * as cdaV0 from '@daml.js/create-daml-app-0.1.0';
+import * as forumV0 from '@daml.js/forum-0.1.0';
+import {MigrateUser} from '@daml.js/migration-v0-v1/lib/Migrate';
+</pre>
+
+Then replace the `login` function in `ui/src/components/LoginScreen.tsx`{{open}} with the following
+implementation:
```
Comment (nemanja-da) [inline upgrading/upgrade-ui/2_ui.md]:
```suggestion
Then replace the `login` function in `ui/src/components/LoginScreen.tsx`{{open}} with the following
implementation that has the necessary UI prompts and ledger queries to perform the update:
```

File: upgrading/upgrade-ui/init_background.sh (hunk 1 above)
Comment (ghost) [reply upgrading/upgrade-ui/init_background.sh]: of course.

File: upgrading/upgrade-ui/1_codegen.md (hunk 2 above)
Comment (ghost) [reply upgrading/upgrade-ui/1_codegen.md]: added the explanation.

File: upgrading/upgrade-ui/1_codegen.md (hunk 3 above)
Comment (ghost) [reply upgrading/upgrade-ui/1_codegen.md]: fixed.

File: upgrading/upgrade-ui/1_codegen.md (hunk 4 above)
Comment (ghost) [reply upgrading/upgrade-ui/1_codegen.md]: changed.

File: upgrading/upgrade-ui/2_ui.md (hunk 6 above)
Comment (ghost) [reply upgrading/upgrade-ui/2_ui.md]: added.

File: upgrading/upgrade-ui/1_codegen.md (hunk 5 above)
Comment (ghost) [reply upgrading/upgrade-ui/1_codegen.md]: I rephrased it a little.

**Pull request merged**